Add docker container for oauth server to fix configuration issues (#561)
NVIDIA/NeMo-Agent-Toolkit: examples/front_ends/simple_auth/README.md

     "id_token":      "…"           // if scope contained openid
   }
   ```

6. The agent stores the tokens and uses the `access_token` in the
   `Authorization: Bearer …` header when invoking tools that need auth.

*Why this flow?*

- Supports **confidential clients** (can keep a secret) *and* public clients with **PKCE**.
- Refresh tokens keep long‑running agents from re‑prompting the user.
- Works across browsers, CLI apps, and UI front‑ends.

## Running the Demo OAuth Provider Locally

In a separate terminal, you can run a demo OAuth 2.0 provider using the [`Authlib`](https://docs.authlib.org/en/latest/)
library. This will allow you to test the OAuth 2.0 Authorization Code Flow with your agent.

### Quick Start with Docker (Recommended)

The easiest way to get started is using Docker, which works seamlessly across all systems (macOS, Windows, Linux):

**Run the example**

```bash

> # From the repository root directory
> docker compose -f examples/front_ends/simple_auth/docker-compose.yml --project-directory examples/front_ends/simple_auth up

Reviewer: Prefer bringup in detached -- otherwise they need to launch another terminal instance.
```suggestion
# From this example's directory
docker compose -f docker-compose.yml up -d
```

```

This will automatically:

- Clone the OAuth2 server example
- Install all dependencies
- Start the server on `http://localhost:5000`
- Set the necessary environment variables for local development

**Note**: The `AUTHLIB_INSECURE_TRANSPORT=1` environment variable is set automatically for local development to allow `http://` callback URLs. This should never be used in production.

Browse to **`http://localhost:5000/`** – you should see the demo home page. Sign up with any name.

**To stop the Docker services:**

```bash

> docker compose -f examples/front_ends/simple_auth/docker-compose.yml --project-directory examples/front_ends/simple_auth down

Reviewer: ```suggestion
# From this example's directory
docker compose -f docker-compose.yml down
```
